attestation-service/src/token/ear_broker.rs

```diff
@@ -0,0 +1,496 @@
+        init_data_claims: serde_json::Value,
+        runtime_data_claims: serde_json::Value,
+        reference_data_map: HashMap<String, Vec<String>>,
+        tee: Tee,
+    ) -> Result<String> {
+
+        let tcb_claims = transform_claims(
+            tcb_claims,
+            init_data_claims.clone(),
+            runtime_data_claims.clone(),
+            tee,
+        )?;
+        debug!("tcb_claims: {:#?}", tcb_claims);
+
+        let tcb_claims_json = serde_json::to_string(&tcb_claims)?;
+
+        let reference_data = json!({
+            "reference": reference_data_map,
+        });
+        let reference_data = serde_json::to_string(&reference_data)?;
+
+        if policy_ids.len() > 1 {
+            warn!("EAR token only accepts the first policy. The rest will be ignored.");
+        }
+
+        if policy_ids.is_empty() {
+            bail!("No policy is given for EAR token generation.");
+        }
+
+        let policy_results = self
+            .policy_engine
+            .evaluate(
+                &reference_data,
+                &tcb_claims_json,
+                &policy_ids[0],
+                &RULES[..],
+            )
+            .await?;
+
+        let mut appraisal = Appraisal::new();
+
+        for (k, v) in &policy_results.rules_result {
+            let claim_value = v.as_i8().context("Policy claim value not i8")?;
+
+            appraisal
+                .trust_vector
+                .mut_by_name(k)
+                .unwrap()
+                .set(claim_value);
+        }
+
+        if !appraisal.trust_vector.any_set() {
+            bail!("At least one policy claim must be set.");
+        }
+
+        appraisal.update_status_from_trust_vector();
+        appraisal.annotated_evidence = tcb_claims;
+        appraisal.policy_id = Some(policy_ids[0].clone());
+
+        // For now, create only one submod, called `cpu`.
+        // We can create more when we support attesting multiple devices at once.
+        let mut submods = BTreeMap::new();
+        submods.insert("cpu".to_string(), appraisal);
+
+        let now = OffsetDateTime::now_utc();
+        let exp = now
+            .checked_add(Duration::minutes(self.config.duration_min))
+            .ok_or(anyhow!("Token expiration overflow."))?;
+
+        let mut extensions = Extensions::new();
+        extensions.register("exp", 4, ExtensionKind::Integer)?;
+        extensions.set_by_name("exp", ExtensionValue::Integer(exp.unix_timestamp()))?;
+
+        let ear = Ear {
+            profile: self.config.profile_name.clone(),
+            iat: now.unix_timestamp(),
+            vid: VerifierID {
+                build: self.config.build_name.clone(),
+                developer: self.config.developer_name.clone(),
+            },
+            raw_evidence: None,
+            nonce: None,
+            submods,
+            extensions,
+        };
+
+        let signed_ear = ear.sign_jwt_pem(Algorithm::ES256, &self.private_key_bytes)?;
```

Comment:
We now supports to embed `jwk` part or `kid` part into the JWT with https://github.com/veraison/rust-ear/pull/29 / https://docs.rs/ear/0.3.0/ear/struct.Ear.html#method.sign_jwt_pem_with_header. Thus we can embed the public key part of `self.private_key_bytes`. In this way

1. if KBS side set `insecure_key = true`. This public jwk can be used to verify the integrity of the jwt.
2. if KBS side set `insecure_key = false`. Some extra trusted root certificate should be set on KBS config side and this public jwk can be used to verify the integrity of the jwt

Reply:
oh didn't see that they fixed that. i will take a look.

Reply:
Ok, I am using the new JWK support and I got rid of the EAR token verifier. Seems to work.

There is a little bit of repeated code to make the JWKs. I wanted to reuse the `pubkey_jwks` function for EAR and CoCo tokens, but there are a few differences that made it slightly tricky. EAR uses EC not RSA, for instance, and the rust-ear crate expects a jsonwebtoken struct rather than a json string. I think we can probably converge these more in a future PR, but for now I kept them separate.

Reply:
Ok. Could you help to mark a TODO comment or an issue to record this?

Reply:
Now we have https://github.com/confidential-containers/trustee/issues/598 and I added a todo.